pyproject.toml

```diff
@@ -0,0 +1,53 @@
+[build-system]
+requires = ["setuptools>=45", "setuptools_scm[toml]>=6.2"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "dpgen2"
+dynamic = ["version"]
+description = "DPGEN2: concurrent learning workflow generating the machine learning potential energy models."
+authors = [
+  {name = "DeepModeling"},
+]
+license = {file = "LICENSE"}
+classifiers = [
+    "Programming Language :: Python :: 3.7",
+    "License :: OSI Approved :: GNU Lesser General Public License v3 (LGPLv3)",
+]
+dependencies = [
+	     'numpy',
+	     'dpdata',
+	     'pydflow>=1.5.10',
+	     'dargs>=0.3.1',
+]
+requires-python = ">=3.7"
+readme = "README.md"
+keywords = ["deep potential", "concurrent learning", "work flow"]
+
+[project.urls]
+Homepage = "https://github.com/deepmodeling/dpgen2"
+documentation = "https://docs.deepmodeling.com/projects/dpgen2"
+repository = "https://github.com/deepmodeling/dpgen2"
+
+[project.entry-points.console_scripts]
+dpgen2 = "dpgen2.entrypoint.main:main"
+
+[project.optional-dependencies]
+docs = [
+    'sphinx',
+    'recommonmark',
+    'sphinx_rtd_theme>=1.0.0rc1',
+    'numpydoc',
+    'm2r2',
+    'deepmodeling-sphinx>=0.1.1',
+    'sphinx-argparse',
+]
+
+[tool.setuptools.packages.find]
+include = ["dpgen2"]
```

Comment:
Use `dpgen2*` to match submodules (e.g. `dpgen2.entrypoint`). Otherwise only `dpgen2` is matched.

Reply:
https://github.com/deepmodeling/dpgen2/blob/8589a93fd929ca3ca553ce300b606af7a36790c9/.github/workflows/test.yml#L22

The current behavior is testing the package without installing it. 
We may:
1. install the package.
2. restore requirement.txt, and find someway of using it in pyproject.toml, just like the what the setup.py does. 
What do you think?

Reply:
I prefer `pip install -e .` in editable mode. There is also a funny workaround in https://github.com/pypa/pip/issues/8049#issuecomment-633845028.